pvlib/test/test_ivtools.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu May  9 10:51:15 2019
+
+@author: cwhanse
+"""
+
+import numpy as np
+import pandas as pd
+import pytest
+from pvlib import pvsystem
+from pvlib import ivtools
+from pvlib.test.conftest import requires_scipy, requires_pysam
+
+
+def get_test_iv_params():
+    return {'IL': 8.0, 'I0': 5e-10, 'Rsh': 1000, 'Rs': 0.2, 'nNsVth': 1.61864}
+
+
+def get_cec_params_cansol_cs5p_220p():
+    return {'V_mp_ref': 46.6, 'I_mp_ref': 4.73, 'V_oc_ref': 58.3,
+            'I_sc_ref': 5.05, 'alpha_sc': 0.0025, 'beta_voc': -0.19659,
+            'gamma_pmp': -0.43, 'cells_in_series': 96}
+
+
+@pytest.fixture()
+def sam_data():
+    return pvsystem.retrieve_sam('cecmod')
+
+
+@pytest.fixture()
+def cec_module_parameters(sam_data):
+    modules = sam_data
+    module = "Canadian_Solar_CS5P_220P"
+    module_parameters = modules[module]
+    return module_parameters
+
+
+@requires_scipy
+def test_fit_sde_sandia():
+    test_params = get_test_iv_params()
+    testcurve = pvsystem.singlediode(photocurrent=test_params['IL'],
+                                     saturation_current=test_params['I0'],
+                                     resistance_shunt=test_params['Rsh'],
+                                     resistance_series=test_params['Rs'],
+                                     nNsVth=test_params['nNsVth'],
+                                     ivcurve_pnts=300)
+    expected = tuple(test_params[k] for k in ['IL', 'I0', 'Rsh', 'Rs',
+                     'nNsVth'])
+    result = ivtools.fit_sde_sandia(v=testcurve['v'], i=testcurve['i'],
+                                    v_oc=testcurve['v_oc'],
+                                    i_sc=testcurve['i_sc'],
+                                    v_mp=testcurve['v_mp'],
+                                    i_mp=testcurve['i_mp'])
+    assert np.allclose(result, expected, rtol=5e-5)
+
+
+@requires_pysam
+def test_fit_cec_sam(cec_module_parameters):
+    sam_parameters = cec_module_parameters
+    cec_list_data = get_cec_params_cansol_cs5p_220p()
+    I_L_ref, I_o_ref, R_sh_ref, R_s, a_ref, Adjust = \
+        ivtools.fit_cec_sam(
+            celltype='polySi', v_mp=cec_list_data['V_mp_ref'],
+            i_mp=cec_list_data['I_mp_ref'], v_oc=cec_list_data['V_oc_ref'],
+            i_sc=cec_list_data['I_sc_ref'], alpha_sc=cec_list_data['alpha_sc'],
+            beta_voc=cec_list_data['beta_voc'],
+            gamma_pmp=cec_list_data['gamma_pmp'],
+            cells_in_series=cec_list_data['cells_in_series'], temp_ref=25)
```

Comment:
I'm trying hard to conform to pvlib with naming in my own code, so looking more closely at this than necessary perhaps.  Do `V_mp_ref` etc. establish a new pattern?  I thought I had detected that lower case `v` was usually followed by an underscore, but not upper case `V`.

Reply:
lower case is preferred for variables. In this case, `V_mp_ref`, `I_sc_ref`, et al. are key values from the column headings in the CEC module parameter data, and we're stuck with them unless we rewrite the io function.